### ISSUE TITLE
EDC 0.6.0 → Azure Deployment→ EnvironmentCredential authentication unavailable

### ISSUE THREAD

**Issue by jerzy-wachowiak** [eclipse-tractusx/tractusx-edc#1150]
EDC 0.6.0 Azure deployment crashes with the log message SEVERE from control-plane:

```
	INFO 2024-03-22T09:13:59.914096456 Initialized OAuth2 Client
	INFO 2024-03-22T09:13:59.915613052 Initialized SSI MIW OAuth2 Client
	SEVERE 2024-03-22T09:14:06.198695378 Error accessing secret ssi-bpnl00000007txd8
	com.azure.identity.CredentialUnavailableException: EnvironmentCredential authentication unavailable. Environment variables are not fully configured.To mitigate this issue, please refer to the troubleshooting guidelines here at https://aka.ms/azsdk/java/identity/environmentcredential/troubleshoot
	WorkloadIdentityCredential authentication unavailable. The workload options are not fully configured. See the troubleshooting guide for more information. https://aka.ms/azsdk/java/identity/workloadidentitycredential/troubleshoot
	Managed Identity authentication is not available.
	Cannot get accounts from token cache. Error: Could not load the brokered authentication library. Ensure that the azure-identity-broker library is on the classpath.
	IntelliJ Authentication not available. Please log in with Azure Tools for IntelliJ plugin in the IDE. Fore more details refer to the troubleshooting guidelines here at https://aka.ms/azsdk/java/identity/intellijcredential/troubleshoot
	AzureCliCredential authentication unavailable. Azure CLI not installed.To mitigate this issue, please refer to the troubleshooting guidelines here at https://aka.ms/azsdk/java/identity/azclicredential/troubleshoot
	Unable to execute PowerShell. Please make sure that it is installed in your system
	AzureDeveloperCliCredential authentication unavailable. Azure Developer CLI not installed.To mitigate this issue, please refer to the troubleshooting guidelines here at https://aka.ms/azsdk/java/identity/azdevclicredential/troubleshootTo mitigate this issue, please refer to the troubleshooting guidelines here at https://aka.ms/azure-identity-java-default-azure-credential-troubleshoot
```

Env. variable for Azure are set in deployment-controlplane.yaml#VAULT:

```
            - name: "EDC_VAULT_CLIENTID"
              value: {{ .Values.vault.azure.client | required ".Values.vault.azure.client is required" | quote  }}
            - name: "EDC_VAULT_TENANTID"
              value: {{ .Values.vault.azure.tenant | required ".Values.vault.azure.tenant is required" | quote }}
            - name: "EDC_VAULT_NAME"
              value: {{ .Values.vault.azure.name | required ".Values.vault.azure.name is required" | quote }}
            # only set the env var if config value not null
            {{- if .Values.vault.azure.secret }}
            - name: "EDC_VAULT_CLIENTSECRET"
              value: {{ .Values.vault.azure.secret | quote }}
             {{- end }}
```

**Comment by jerzy-wachowiak** [comment]
This worked for me: `deployment-controlplane.yaml`

```
            ###########
            ## VAULT ##
            ###########

            - name: "EDC_VAULT_CLIENTID"
              value: {{ .Values.vault.azure.client | required ".Values.vault.azure.client is required" | quote  }}
            - name: "EDC_VAULT_TENANTID"
              value: {{ .Values.vault.azure.tenant | required ".Values.vault.azure.tenant is required" | quote }}
            - name: "EDC_VAULT_NAME"
              value: {{ .Values.vault.azure.name | required ".Values.vault.azure.name is required" | quote }}
            # only set the env var if config value not null
            {{- if .Values.vault.azure.secret }}
            - name: "EDC_VAULT_CLIENTSECRET"
              value: {{ .Values.vault.azure.secret | quote }}
             {{- end }}
            # only set the env var if config value not null
            {{- if .Values.vault.azure.certificate }}
            - name: "EDC_VAULT_CERTIFICATE"
              value: {{ .Values.vault.azure.certificate | quote }}
            {{- end }}
            # Bug fix https://github.com/eclipse-tractusx/tractusx-edc/issues/1150
            - name: "AZURE_CLIENT_ID"
              value: {{ .Values.vault.azure.client | required ".Values.vault.azure.client is required" | quote  }}
            - name: "AZURE_TENANT_ID"
              value: {{ .Values.vault.azure.tenant | required ".Values.vault.azure.tenant is required" | quote }}
            - name: "AZURE_CLIENT_SECRET"
              value: {{ .Values.vault.azure.secret | quote }}
```

**Comment by dreni** [comment]
a quick fix if you have to use the edc-tractusx helm chart, is to add the missing env variables:
controlplane:
    # quick and dirty fix: https://github.com/eclipse-tractusx/tractusx-edc/issues/1150
    env:
      AZURE_CLIENT_ID: 
      AZURE_CLIENT_SECRET: 
      AZURE_TENANT_ID:

**Comment by dreni** [comment]
> This worked for me: `deployment-controlplane.yaml`
> 
> ```
>             ###########
>             ## VAULT ##
>             ###########
> 
>             - name: "EDC_VAULT_CLIENTID"
>               value: {{ .Values.vault.azure.client | required ".Values.vault.azure.client is required" | quote  }}
>             - name: "EDC_VAULT_TENANTID"
>               value: {{ .Values.vault.azure.tenant | required ".Values.vault.azure.tenant is required" | quote }}
>             - name: "EDC_VAULT_NAME"
>               value: {{ .Values.vault.azure.name | required ".Values.vault.azure.name is required" | quote }}
>             # only set the env var if config value not null
>             {{- if .Values.vault.azure.secret }}
>             - name: "EDC_VAULT_CLIENTSECRET"
>               value: {{ .Values.vault.azure.secret | quote }}
>              {{- end }}
>             # only set the env var if config value not null
>             {{- if .Values.vault.azure.certificate }}
>             - name: "EDC_VAULT_CERTIFICATE"
>               value: {{ .Values.vault.azure.certificate | quote }}
>             {{- end }}
>             # Bug fix https://github.com/eclipse-tractusx/tractusx-edc/issues/1150
>             - name: "AZURE_CLIENT_ID"
>               value: {{ .Values.vault.azure.client | required ".Values.vault.azure.client is required" | quote  }}
>             - name: "AZURE_TENANT_ID"
>               value: {{ .Values.vault.azure.tenant | required ".Values.vault.azure.tenant is required" | quote }}
>             - name: "AZURE_CLIENT_SECRET"
>               value: {{ .Values.vault.azure.secret | quote }}
> ```

since the helm chat "tractusx-connector-azure-vault" is azure specific, I would replace the following env vars:
EDC_VAULT_CLIENTID -> AZURE_CLIENT_ID
EDC_VAULT_TENANTID -> AZURE_TENANT_ID
EDC_VAULT_CLIENTSECRET -> AZURE_CLIENT_SECRET

**Comment by wolf4ood** [comment]
This has been already fixed and available in 0.7.x

https://github.com/eclipse-tractusx/tractusx-edc/pull/1185
